### PR TITLE
Drp2 1167 download to csv

### PR DESCRIPTION
**Pull request by rossdowthwaite**
Download report data as CSV file: https://dsdmoj.atlassian.net/jira/software/c/projects/DPR2/boards/1345?quickFilter=4664&selectedIssue=DPR2-1167

See [this file](https://github.com/ministryofjustice/hmpps-digital-prison-reporting-frontend/pull/268/files#diff-6d02e64464ca36ff032c5103086895692a8706f659172e6284acc649d068e478R7) for a description of the process

### CODE REVIEW

File: src/dpr/routes/asyncReports.ts [hunk 1]
```diff
@@ -244,7 +244,7 @@ export default function routes({
   // 3 - VIEw REPORT
   const viewReportPaths = [
     '/async/:type/:reportId/:id/request/:tableId/report',
-    '/async/:type/:reportId/:id/request/:tableId/report/download',
+    '/async/:type/:reportId/:id/request/:tableId/report/:download',
```
Comment (stalwartstu) [inline src/dpr/routes/asyncReports.ts:247]: @MOJ-Fordie The "/download" URL search in App Insights will need to change to "/downloadReport/" once this change goes live.

File: src/dpr/utils/downloadUtils.ts [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    const userId = res.locals.user?.uuid ? res.locals.user.uuid : 'userId'
+    const token = res.locals.user?.token ? res.locals.user.token : 'token'
+
+    const { reportId, id, tableId, dataProductDefinitionsPath, reportName, variantName, cols: columns } = req.body
+
+    const canDownload = await services.downloadPermissionService.downloadEnabled(userId, reportId, id)
+    if (!canDownload) {
+      res.redirect(`/async/report/${reportId}/${id}/request/${tableId}/report/download-disabled`)
+    } else {
+      let reportData = await services.reportingService.getAsyncReport(token, reportId, id, tableId, {
```
Comment (stalwartstu) [inline src/dpr/utils/downloadUtils.ts:44]: It also looks like summary and section reports will just download the raw source data. Is that an expected behaviour? Or should it be added to the assumptions?

File: test-app/server.js [hunk 3]
```diff
@@ -144,9 +145,12 @@ const mockUserStore = new MockUserStoreService()
 const requestedReportService = new RequestedReportService(mockUserStore)
 const recentlyViewedService = new RecentlyViewedStoreService(mockUserStore)
 const bookmarkService = new BookmarkService(mockUserStore)
+const downloadPermissionService = new DownloadPermissionService(mockUserStore)
```
Comment (stalwartstu) [inline test-app/server.js:148]: Will users of our library need to set this up too? If so, please could you add it to the Integration guide?

**Review by stalwartstu — APPROVED** [review]

File: test-app/server.js (hunk 3 above)
Comment (rossdowthwaite) [reply test-app/server.js:148]: Theres been no talk of download to the sync journey. there currently no message or feedback for in the sync journey. Perhaps its been forgotten about

**Pull request merged**